state/event_table.go

```diff
@@ -336,6 +338,30 @@ func (t *EventTable) LatestEventNIDInRooms(txn *sqlx.Tx, roomIDs []string, highe
 	return
 }
 
+func (t *EventTable) Redact(txn *sqlx.Tx, roomVer string, eventIDs []string) error {
+	// verifyAll=false so if we are asked to redact an event we don't have we don't fall over.
+	eventsToRedact, err := t.SelectByIDs(txn, false, eventIDs)
+	if err != nil {
+		return fmt.Errorf("EventTable.Redact[%v]: %s", eventIDs, err)
+	}
+	rv, err := gomatrixserverlib.GetRoomVersion(gomatrixserverlib.RoomVersion(roomVer))
+	if err != nil {
+		// unknown room version... let's just default to "1"
```

Comment:
Worth a warning? Ditto for the case in the accumulator where you fall back to 1

Reply:
This can and will happen for unknown MSC room versions, so yeah warning but not sentry.